src/reynir_correct/pattern.py

```diff
@@ -470,6 +849,171 @@ def create_patterns(cls) -> None:
+                    "VP > { VP > { 'leiða' } NP-PRD > { ('líkur' | 'rök') } PP > { P > { 'af' } } } ",
+                    cls.wrong_preposition_leida_af,
+                    None,
+                )
+            )
+            # Catch "Hann hefur aldrei leitt líkur af því."
+            p.append(
+                (
+                    "líkur",  # Trigger lemma for this pattern
+                    "VP > { VP >> { VP > { 'leiða' } NP > { 'líkur' } } PP > { 'af' } } ",
+                    cls.wrong_preposition_leida_af,
+                    None,
+                )
+            )
+            # Catch "Tíminn markar (ekki) upphaf af því."
+            p.append(
+                (
+                    "upphaf",  # Trigger lemma for this pattern
+                    "VP > { VP > { 'marka' } NP-OBJ > { 'upphaf' PP > { 'af' } } }",
+                    cls.wrong_preposition_marka_af,
+                    None,
+                )
+            )
+            # Catch "Það markar (ekki) upphaf af því."
+            p.append(
+                (
+                    "upphaf",  # Trigger lemma for this pattern
+                    "VP > { VP > { VP > { 'marka' } NP-SUBJ > { 'upphaf' } } PP > { 'af' } }",
+                    cls.wrong_preposition_marka_af,
+                    None,
+                )
+            )
+
+            # Catch "Jón leggur hann af velli."
+            p.append(
+                (
+                    "leggja",  # Trigger lemma for this pattern
+                    "VP > { VP > { VP > { 'leggja' } } PP > { P > { 'af' } NP > { 'völlur' } } }",
+                    cls.wrong_preposition_leggja_af,
+                    None,
+                )
+            )
+            # Catch "Jón hefur lagt hann af velli."
+            p.append(
+                (
+                    "leggja",  # Trigger lemma for this pattern
+                    "VP > { VP > { VP > { VP > { 'leggja' } } } PP > { P > 'af' NP > 'völlur' } }",
+                    cls.wrong_preposition_leggja_af,
+                    None,
+                )
+            )
+            # Catch "Jón hefur ekki lagt hann af velli."
+            p.append(
+                (
+                    "leggja",  # Trigger lemma for this pattern
+                    "VP > { VP > { VP > { VP > { VP > { 'leggja' } } } } PP > { P > { 'af' } NP > { 'völlur' } } }",
+                    cls.wrong_preposition_leggja_af,
+                    None,
+                )
+            )
+            # Catch "Jón kann það (ekki) utan af."
+            p.append(
+                (
+                    "kunna",  # Trigger lemma for this pattern
+                    "VP > { VP > { 'kunna' } ADVP > { 'utan' } ADVP > { 'af' } }",
+                    cls.wrong_preposition_utan_af,
+                    None,
+                )
+            )
+            # Catch "Honum varð af ósk sinni."
+            p.append(
+                (
+                    "verða",  # Trigger lemma for this pattern
```

Comment:
Use 'ósk' as a trigger lemma? Much less common than 'verða'.

Reply:
Yes, that makes sense